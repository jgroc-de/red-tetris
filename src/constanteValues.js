export const numberOfLines = 20
export const numberOfColumns = 10
export const filledSquared = '-'
export const filledTetriminosSquared = 'x'
export const emptySquared = ' '
export const rotationCases = 4
export const LShape = [
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const JShape = [
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ]
]
export const SShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ],
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const OShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
]
export const TShape = [
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const ZShape = [
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],

]
export const IShape = [
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
    ],
]
