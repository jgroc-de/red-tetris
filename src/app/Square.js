import React from 'react'
import {filledSquared, filledTetriminosSquared} from '../constanteValues'

function Square(props) {
    let cssClasses = ['square']
    if (props.square === filledTetriminosSquared) {
        cssClasses.push('colored')
    } else if (props.square === filledSquared) {
        cssClasses.push('colored-grey')
    }
    return (<div className={cssClasses.join(' ')}></div>)
}

export default Square
