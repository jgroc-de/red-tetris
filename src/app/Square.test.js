import React from 'react'
import renderer from 'react-test-renderer'
import Square from './Square'
import  {filledTetriminosSquared, filledSquared} from '../constanteValues'

test('empty square', () => {
    const component = renderer.create(<Square square={null} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('filled square', () => {
    const component = renderer.create(<Square square={filledSquared} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('tetriminos square', () => {
    const component = renderer.create(<Square square={filledTetriminosSquared} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
