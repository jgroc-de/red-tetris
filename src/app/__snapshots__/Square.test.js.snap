// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty square 1`] = `
<div
  className="square"
/>
`;

exports[`filled square 1`] = `
<div
  className="square colored-grey"
/>
`;

exports[`tetriminos square 1`] = `
<div
  className="square colored"
/>
`;
